batchlight/src/main/java/co/yello/db/batchlight/android/AndroidDelegate.kt

```diff
@@ -0,0 +1,25 @@
+package co.yello.db.batchlight.android
+
+import co.yello.db.batchlight.sqlAndroidPreparedStatementStartIndex
+
+class SqliteDelegate(
+      fieldsPerItem: Int,
+      maxFields: Int
+) {
+    /**
+     * The maximum number of items that can be inserted per statement.
+     */
+    val maxInsertSize = if (fieldsPerItem > 0) maxFields / fieldsPerItem else 1
+
+    /**
+     * For a SQLite insert statement each item has the format (?, ?, ?)
+     */
+    val objectStatement = (1..fieldsPerItem).joinToString(
+        prefix = "(",
+        transform = { "?" },
+        separator = ",",
+        postfix= ")"
```

Comment:
It doesn't *really* matter, but why is transform surrounded by brackets and the others are not? This is just out of my own curiosity! Everything else looks good.

Reply:
Ill refactor that

Reply:
After looking into it, the transform actually takes in a closure so that is what the brackets are for. All other params are type CharSequence.